pkg/planner/core/point_get_plan.go

```diff
@@ -883,7 +883,7 @@ func TryFastPlan(ctx base.PlanContext, node ast.Node) (p base.Plan) {
 	ctx.GetSessionVars().PlanColumnID.Store(0)
 	switch x := node.(type) {
 	case *ast.SelectStmt:
-		if x.SelectIntoOpt != nil {
+		if x.SelectIntoOpt != nil || x.LockInfo != nil {
```

Comment:
The lock operations of point get and batch point get are push down. The original design of selectLockExec was only for cop reads that cannot be pushed down.

`TryFastPlan` is designed be used to generate point/batch point get plans.

The issue is not a high priority one by now, better to resolve the plan display issue after some proper executor builder and transaction state refactor.

Reply:
so it's barely a explained issue, not the correctness one, right?

Reply:
Yes the correctness one is https://github.com/pingcap/tidb/issues/54652. 
The severity of #54705 is moderate, there is also related behaviour fix checking `lock is needed or not` in https://github.com/pingcap/tidb/pull/54738/files#diff-359d434f5527f2d2157554bb9081bdc497f25e0a76253907fb5c9d1bdfb9b233L770.